torch/csrc/jit/codegen/cuda/test/test_gpu3.cpp

```diff
@@ -6523,7 +6523,7 @@ TEST_F(NVFuserTest, FusionExpandedInput_CUDA) {
   fusion->addOutput(tv1);
 
   auto options = at::TensorOptions().dtype(kFloat).device(at::kCUDA, 0);
-  at::Tensor t0 = at::randn({4096, 1, 3}, options).expand({-1, 7, -1});
+  at::Tensor t0 = at::randn({4096, 1, 4}, options).expand({-1, 7, -1});
```

Comment:
Changing the last dimension to `4` triggers a vectorization of 4. Indeed, there are actually two problems:
1. Even if the last dim is 3, we should still vectorize. But our scheduler is not smart enough to analyze this case and vectorize.
2. `getVectorizedFusionInputOutput` should't have the assert.
This PR only fixes the second.

Reply:
There are so many issues with vectorization analysis and validation :(.

Reply:
@naoyam For issue 1, I think we really need to change the semantics of contiguity to ignore broadcasting to fix it. Otherwise, there is no way for `expandVectorizationToContigMergedDomains` to know if the last dim can be expanded left.